context.go

```diff
@@ -0,0 +1,69 @@
+package api2go
+
+import (
+	"golang.org/x/net/context"
+	"time"
+)
+
+// APIContext api2go context for handlers, nil implementations related to Deadline and Done.
+type APIContext struct {
```

Comment:
I think @sharpner means that the api2go `APIContext` should have an additional interface, so that a user can extend/change the `APIContext` easily. And yes, it must be public in order to be jused by somebody that does not want to implement it by himself. But all our internal methods would require the interface. That makes sense or?